ci_framework/hooks/playbooks/fetch_compute_facts.yml

```diff
@@ -0,0 +1,35 @@
+---
+- hosts: all
+  gather_facts: true
+  tasks:
+    - name: Controller stores facts info into parameters file
+      when: "inventory_hostname == 'controller'"
+      block:
+        - name: Set facts for further usage within the framework
+          vars:
+            isolated_net_regex: "^192\\.168\\.0\\..*"
+          ansible.builtin.set_fact:
+            cifmw_external_compute_info:
+              remote_key: "{{ ansible_user_dir }}/.ssh/id_rsa"
+              private_net_ip: >-
+                {{
+                  hostvars.compute.ansible_all_ipv4_addresses |
+                  select('match', isolated_net_regex) |
+                  first
+                }}
+              remote_user: "{{ hostvars.compute.ansible_env.USER }}"
+            cacheable: true
+
+        - name: Create artifacts dir
+          ansible.builtin.file:
+            path: "{{ ansible_user_dir }}/ci-framework-data/artifacts/parameters/"
+            state: directory
+
+        - name: Save compute info
+          vars:
+            file_content:
+              test_dumped_vars: "{{ hostvars }}"
+              cifmw_external_compute_info: "{{ cifmw_external_compute_info }}"
+          ansible.builtin.copy:
+            dest: "{{ ansible_user_dir }}/ci-framework-data/artifacts/parameters/external-compute-parameters.yml"
```

Comment:
using this:
https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/run_hook/tasks/playbook.yml#L74-L84

`run_hook` is already allowing to load generated content from within a hook.

This mean, basically, that if you name the "dest" as follow: `{ cifmw_basedir }}/artifacts/{{ step }}_{{ hook_name }}.yml`

it will be loaded directly after the hook is finished, exposing the parameter you create in the "deploy-edpm.yml" playbook run.

This means that L23-26 are useless (and potentially dangerous since it may point to another location), and that you probably don't need to add L21 (cacheable: true).